ruby/app.rb

```diff
@@ -176,36 +204,56 @@
 # rubocop:disable Metrics/BlockLength
 get '/api/assets' do
   begin
+    options = {
+      client_report_id: '123',
+      webhook: 'https://www.example.com',
+      user: {
+        client_user_id: '789',
+        first_name: 'Jane',
+        middle_name: 'Leah',
+        last_name: 'Doe',
+        ssn: '123-45-6789',
+        phone_number: '(555) 123-4567',
+        email: 'jane.doe@example.com'
+      }
+    }
+    asset_report_create_request = Plaid::AssetReportCreateRequest.new
+    asset_report_create_request.access_tokens = [access_token]
+    asset_report_create_request.days_requested = 20
+    asset_report_create_request.options = options
     asset_report_create_response =
-      client.asset_report.create([access_token], 10, {})
+      client.asset_report_create(asset_report_create_request)
     pretty_print_response(asset_report_create_response)
-  rescue Plaid::PlaidAPIError => e
+  rescue Plaid::ApiError => e
     error_response = format_error(e)
     pretty_print_response(error_response)
     content_type :json
     error_response.to_json
   end
 
-  asset_report_token = asset_report_create_response['asset_report_token']
-
+  asset_report_token = asset_report_create_response.asset_report_token
   asset_report_json = nil
-  num_retries_remaining = 20
-  while num_retries_remaining > 0
-    begin
-      asset_report_get_response = client.asset_report.get(asset_report_token)
-      asset_report_json = asset_report_get_response['report']
-      break
-    rescue Plaid::PlaidAPIError => e
-      if e.error_code == 'PRODUCT_NOT_READY'
-        num_retries_remaining -= 1
-        sleep(1)
-        next
+  asset_report_pdf = nil
+
+  begin
+    50.times do
+      begin
+        asset_report_get_request = Plaid::AssetReportGetRequest.new
+        asset_report_get_request.asset_report_token = asset_report_token
+        asset_report_get_response = client.asset_report_get(asset_report_get_request)
+        asset_report_json = asset_report_get_response.report
+        asset_report_pdf_get_request = Plaid::AssetReportPDFGetRequest.new
+        asset_report_pdf_get_request.asset_report_token = asset_report_token
+        asset_report_pdf = client.asset_report_pdf_get( asset_report_pdf_get_request)
+        break
+      rescue Plaid::ApiError => e
+        json_response = JSON.parse(e.response_body)
+        raise e if json_response['error_code'] != 'PRODUCT_NOT_READY'
+
+        sleep 1
       end
-      error_response = format_error(e)
-      pretty_print_response(error_response)
-      content_type :json
-      return error_response.to_json
```

Comment:
the new code seems substantially different than the old code, could you explain the change a little? Where did the pretty printing the error go? Where fo we return the error now?